File: No.121_BestTimeBuyAndSellStock/No.121_BestTimeBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        
        if(prices.length == 0)
            return 0;        
        
        int minPrice = Integer.MAX_VALUE;
        int profit = 0;
        for(int i = 0 ; i < prices.length ; i++)
        {
            if(minPrice >= prices[i])
                minPrice = prices[i];
            else
            {
                if((prices[i] - minPrice) > profit)
                    profit = prices[i] - minPrice;
            }
        }
        return profit;
    }
}

File: No.121_BestTimeBuyAndSellStock/No.121_BestTimeBuyAndSellStock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        if len(prices) == 0:
            return 0
        
        minPrice = sys.maxsize
        profit = 0
        for p in prices:
            if minPrice >= p:
                minPrice = p
            else:
                if p - minPrice > profit:
                    profit = p - minPrice
        
        return profit